src/backend/tools/gmail/tool.py

```diff
@@ -0,0 +1,78 @@
+from typing import Any, Dict, List
+
+from backend.config.settings import Settings
+from backend.crud import tool_auth as tool_auth_crud
+from backend.schemas.tool import ToolCategory, ToolDefinition
+from backend.services.logger.utils import LoggerFactory
+from backend.tools.base import BaseTool
+from backend.tools.gmail.auth import GmailAuth
+from backend.tools.gmail.constants import GMAIL_TOOL_ID, SEARCH_LIMIT
+from backend.tools.gmail.utils import get_gmail_service
+
+logger = LoggerFactory().get_logger()
+
+
+class GmailTool(BaseTool):
+    """
+    Tool that searches Gmail for messages and files based on a query.
+    """
+
+    ID = GMAIL_TOOL_ID
+    CLIENT_ID = ""
+    CLIENT_SECRET = ""
+
+    @classmethod
+    def is_available(cls) -> bool:
+        settings = Settings()
```

Comment:
Same thing here, let's update Settings() to use `.get()`

Reply:
Please also move the attribute setting out of `is_available` and directly onto the class attribute, then only check that they're not None in `is_available`